### PR TITLE
feat: saved credit card

### PR DESCRIPTION
**Pull request by fandihungryhub**

### CODE REVIEW

File: api/credit_card/createCreditCard.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import useHttp from "~/composables/useHttp";
+import { CreditCard, CreditCardData } from "~/types/CreditCard";
+
+type CCPayload = {
+  access_token: string;
```
Comment (afif-hh) [inline api/credit_card/createCreditCard.ts:5]: harunys camelCase

Comment (fandihungryhub) [reply api/credit_card/createCreditCard.ts]: updated

**Review by afif-hh — CHANGES_REQUESTED** [review]

File: components/booking/Payment/CreditCardGuest.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        </template>
+      </BottomSheet>
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, computed, onMounted } from "vue";
+import { storeToRefs } from "pinia";
+import { useCreditCardStore } from "~~/stores/creditCard";
```
[the remark this answers is not in the document]
Comment (fandihungryhub) [reply components/booking/Payment/CreditCardGuest.vue]: updated

File: components/booking/Payment/CreditCardGuest.vue [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    securityCode: ccModel.value.ccSecurityCode,
+    logo: "",
+  };
+  showBottomSheet.value = false;
+  emit("on-save");
+}
+
+function onCancel() {
+  //creditCardStore.resetCCModel();
+  /*paymentMethod.value = "";
```
[the remark this answers is not in the document]
Comment (fandihungryhub) [reply components/booking/Payment/CreditCardGuest.vue]: updated

**Review by afif-hh — APPROVED** [review]

**Pull request merged**